src/plugins/embeddable/public/lib/containers/embeddable_child_panel.tsx

```diff
@@ -11,20 +11,30 @@ import React from 'react';
 
 import { EuiLoadingChart } from '@elastic/eui';
 import { Subscription } from 'rxjs';
+import { distinct, tap, map } from 'rxjs/operators';
 import { ErrorEmbeddable, IEmbeddable } from '../embeddables';
 import { IContainer } from './i_container';
 import { EmbeddableStart } from '../../plugin';
+import { EmbeddableError, EmbeddableOutput } from '../embeddables/i_embeddable';
+
+export type EmbeddableRenderStatus = 'loading' | 'loaded' | 'rendered' | 'error';
```

Comment:
Rendering is only one possible status so maybe the type name should be more generic? E.g.:

* `EmbeddableSetupStatus`
* `EmbeddableSetupPhase`
* `EmbeddableInitStatus`

Also, do we envision that this can also be used for top-level components, not only embeddables? In that case it would make sense to eliminate the word `Embeddable` from the type name.